### PR TITLE
Fix global gene search for studies with default numeric annotation (SCP-4200)

### PR DESCRIPTION
**Pull request by eweitz**
This fixes the display of results for global gene search when the study has a default annotation of type "numeric".

Previously, likely for at least 1-2 years, an error was displayed for such results.  Now, an annotated scatter plot is shown, similar to what's shown in gene search for the Explore tab in Study Overview (#1435).

Here's how it looks:

https://user-images.githubusercontent.com/1334561/161144590-a0d6c4f2-6c27-4f77-9566-69766fb9f524.mov

This satisfies SCP-4200.

### CODE REVIEW

File: app/javascript/components/search/genes/StudyGeneExpressions.jsx [hunk 1]
```diff
@@ -17,7 +18,7 @@ import { getDefaultClusterParams, getAnnotationValues, emptyDataParams } from '~
   */
 export default function StudyGeneExpressions({ study }) {
   const [clusterParams, setClusterParams] = useState(_clone(emptyDataParams))
-  const [annotationList, setAnnotationList] = useState(null)
+  const [annotationList, setAnnotationList] = useState(study.annotation_list ?? null)
   let controlClusterParams = _clone(clusterParams)
```
Comment (devonbush) [inline app/javascript/components/search/genes/StudyGeneExpressions.jsx:23]: Add a `getAnnotationFromIdentifier` helper method to cluster-utils.js that mirrors the existing `getIdentifierFromAnnotation`, and then you can do
```suggestion
const defaultAnnotation = getAnnotationFromIdentifier(study.default_annotation)
const defaultAnnotationIsNumeric = defaultAnnotation?.type === 'numeric'
```
For bonus points, update line 80 of ExploreTabRouter.jsx to use the new helper function

**Review by devonbush — APPROVED** [review]
I'd lean toward sending the whole default annotation identifier just because it seems cleaner than extracting a single boolean, given that we'll probably want to know other things about the default annotation later.  But I don't have my heart set on it :)

**Comment by eweitz** [comment]
To give adequate review time, I'll keep this as originally scoped and open a separate PR when the other fix is ready.

**Review by ehanna4 — CHANGES_REQUESTED** [review]
I think this change might have broken a previous fix for another type of error.
When I run a gene search on `development` 2 studies do not display a graph but instead display a small message explaining why no graph is shown. However, when I run this branch now I see a 'Something went wrong.' red error message
Before (on development branch):
Screen Shot 2022-04-08 at 10 42 59 AM
With this branch:
Screen Shot 2022-04-08 at 10 44 12 AM
Screen Shot 2022-04-08 at 10 41 55 AM

**Comment by eweitz** [comment]
Good catch, @ehanna4!  Fixed in 8eb26f2.  The fix now accounts for studies that lack visualizations.

**Review by jlchang — APPROVED** [review]
Functional review: 
- confirmed annotated scatter generated for studies with default annotation of TYPE numeric 
- confirmed no plot generated for studies with gene expression that have no cluster file and instead the appropriate message "This study does not have cluster data to support visualization in the portal" is conveyed.

**Review by ehanna4 — APPROVED** [review]
Nice, confirmed previous bug is gone and no new issues 👍

**Pull request merged**